controller/cache/cache.go

```diff
@@ -62,13 +71,22 @@ var (
 	// clusterCacheListPageSize is the page size when performing K8s list requests.
 	// 500 is equal to kubectl's size
 	clusterCacheListPageSize int64 = 500
+
+	// clusterCacheRetryLimit sets a retry limit for failed requests during cluster cache sync
+	// If set to 1, retries are disabled.
+	clusterCacheRetryLimit int32 = 1
```

Comment:
The tiniest of nitpicks: I'd expect "retry limit" to mean "limit of the number of retries" where the 2nd request is the first retry. In other words, I'd expect retryLimit=0 to mean "no retries."

I think if we wanted 1 to mean "no retries," we should call the env var "attempt limit."

Reply:
That's actually a good suggestion, thanks :+1: Will change the naming. Although the appropriate changes have been named in GitOps engine like this already :\